internal/plugin.go

```diff
@@ -1621,6 +1627,79 @@ func (p *Plugin) createPlacement(
 	return
 }
 
+// generateSelector determines the type of input and creates a map of selectors to be used in either the
+// clusterSelector or labelSelector field
+func (p *Plugin) generateSelector(
+	resolvedSelectors map[string]interface{},
+) (map[string]interface{}, error) {
+	if resolvedSelectors == nil {
+		return map[string]interface{}{"matchExpressions": []interface{}{}}, nil
+	}
+
+	// Ensure legacy selectors and formatted YAML selectors are not both in use
+	selectorsUsed := map[string]bool{
+		"legacy":       false,
+		"fullSelector": false,
+	}
+
+	for key := range resolvedSelectors {
+		if key == "matchExpressions" || key == "matchLabels" {
+			selectorsUsed["fullSelector"] = true
+		} else {
+			selectorsUsed["legacy"] = true
+		}
+	}
+
+	if selectorsUsed["legacy"] && selectorsUsed["fullSelector"] {
+		return nil, fmt.Errorf("cannot use both key/value pairs and formatted YAML for placement selectors")
+	}
```

Comment:
Is this code needed since you detect if it's a legacy format in `for label, value := range resolvedSelectors {`?